demos/guide/quic-hq-interop-server.c

```diff
@@ -0,0 +1,689 @@
+        fprintf(stderr, "Unable to create BIO_ADDR\n");
+        goto err;
+    }
+
+    /*
+     * Build a INADDR_ANY BIO_ADDR
+     */
+    if (!BIO_ADDR_rawmake(addr, AF_INET, &ina, sizeof(ina), htons(port))) {
+        fprintf(stderr, "unable to bind to port %d\n", port);
+        goto err;
+    }
+
+    /* Bind to the new UDP socket */
+    if (!BIO_bind(fd, addr, 0)) {
+        fprintf(stderr, "cannot bind to %u\n", port);
+        goto err;
+    }
+
+    /*
+     * Create a new datagram socket
+     */
+    sock = BIO_new(BIO_s_datagram());
+    if (sock == NULL) {
+        fprintf(stderr, "cannot create dgram bio\n");
+        goto err;
+    }
+
+    /*
+     * associate the underlying socket with the dgram BIO
+     */
+    if (!BIO_set_fd(sock, fd, BIO_CLOSE)) {
+        fprintf(stderr, "Unable to set fd of dgram sock\n");
+        goto err;
+    }
+
+    /*
+     * Free our allocated addr
+     */
+    BIO_ADDR_free(addr);
+    return sock;
+
+err:
+    BIO_ADDR_free(addr);
+    BIO_free(sock);
+    BIO_closesocket(fd);
+    return NULL;
+}
+
+/**
+ * @brief Processes a new incoming QUIC stream for an HTTP/0.9 GET request.
+ *
+ * This function reads an HTTP/0.9 GET request from the provided QUIC stream,
+ * retrieves the requested file from the server's file system, and sends the
+ * file contents back to the client over the stream.
+ *
+ * @param Pointer to the SSL object representing the QUIC stream.
+ *
+ * Operation:
+ * - Reads the HTTP/0.9 GET request from the client.
+ * - Parses the request to extract the requested file name.
+ * - Constructs the file path using the `fileprefix` directory.
+ * - Reads the requested file in chunks and sends it to the client.
+ * - Concludes the QUIC stream once the file is fully sent.
+ *
+ * Error Handling:
+ * - If the request is invalid or the file cannot be opened, appropriate
+ *   error messages are logged, and the function exits without sending data.
+ * - Errors during file reading or writing to the stream are handled, with
+ *   retries for buffer-related issues (e.g., full send buffer).
+ *
+ * Notes:
+ * - The request is expected to be a valid HTTP/0.9 GET request.
+ * - File paths are sanitized to prevent path traversal vulnerabilities.
+ * - The function uses blocking operations for reading and writing data.
+ *
+ * Usage:
+ * - Called for each accepted QUIC stream to handle client requests.
+ */
+static void process_new_stream(SSL *stream)
+{
+    unsigned char buf[BUF_SIZE];
+    char path[BUF_SIZE];
+    char *req = (char *)buf;
+    char *reqname;
+    char *creturn;
+    size_t nread;
+    BIO *readbio;
+    size_t bytes_read = 0;
+    size_t bytes_written = 0;
+    size_t offset = 0;
+    int rc;
+
+    memset(buf, 0, BUF_SIZE);
+    if (SSL_read_ex(stream, buf, sizeof(buf) - 1, &nread) <= 0)
+        return;
+
+    /* We should have a valid http 0.9 GET request here */
+    fprintf(stderr, "Request is %s\n", req);
+
+    /* Look for the first '/' char in the request */
```

Comment:
nit there should be last '/', but I'm happy to stick the the _basename_ path component. thanks for addressing this.